trade-finance-manager-ui/server/controllers/helpers/amendments.helper.show-amendment-button.test.js

```diff
@@ -0,0 +1,100 @@
+import { TEAM_IDS, DEAL_STATUS } from '@ukef/dtfs2-common';
+import { showAmendmentButton } from './amendments.helper';
+import { AmendmentButtonDealBuilder } from '../../test-mocks/show-amendments-button-deal-builder';
+import { DEAL } from '../../constants';
+
+describe('amendments helper', () => {
+  describe('showAmendmentButton()', () => {
+    const defaultUserTeams = [TEAM_IDS.PIM];
+
+    it.each(getSuccessTestCases())('should return true when $description', ({ deal, userTeams }) => {
+      const result = showAmendmentButton(deal, userTeams);
+      expect(result).toEqual(true);
+    });
+
+    it.each(getFailureTestCases())('should return false when $description', ({ deal, userTeams }) => {
+      const result = showAmendmentButton(deal, userTeams);
+      expect(result).toEqual(false);
+    });
+
+    /**
+     * The builder creates a valid deal object for the amendment button.
+     * The success tests set a specific variable to ensure that given a deal is otherwise valid,
+     * when the testing variable is changed, the button is shown.
+     */
+    function getSuccessTestCases() {
+      return [
+        {
+          description: 'submission type is AIN',
+          deal: new AmendmentButtonDealBuilder().withDealSnapshotSubmissionType(DEAL.SUBMISSION_TYPE.AIN).build(),
+          userTeams: defaultUserTeams,
+        },
+        {
+          description: 'submissionType is MIN',
+          deal: new AmendmentButtonDealBuilder().withDealSnapshotSubmissionType(DEAL.SUBMISSION_TYPE.MIN).build(),
+          userTeams: defaultUserTeams,
+        },
+        {
+          description: 'deal stage is confirmed',
+          deal: new AmendmentButtonDealBuilder().withTfmStage(DEAL.DEAL_STAGE.CONFIRMED).build(),
+          userTeams: defaultUserTeams,
+        },
+        {
+          description: 'deal stage is amendment in progress',
+          deal: new AmendmentButtonDealBuilder().withTfmStage(DEAL.DEAL_STAGE.AMENDMENT_IN_PROGRESS).build(),
+          userTeams: defaultUserTeams,
+        },
```

Comment:
Can we use constants in the descriptions where possible, instead of "in progress", "MIN" strings etc?